### PR TITLE
Add docker-compose support

### PR DESCRIPTION
**Pull request by Lensual**
#81

### CODE REVIEW

**Comment by Lensual** [comment]
目前 playground 依赖的 `NEXT_PUBLIC_REQUEST_URL` 环境变量只在编译时有效，修改后需要重新 `docker compose build`

next框架不太熟悉

**Comment by cyfyifanchen** [comment]
I had a brief discussion with the other engineers yesterday. We believe that most web developers should already be familiar with how `node` and `nvm` work, which is why we’re not using `docker compose build`. 

However, @Lensual, your point is still valid for developers who aren’t as familiar with web development tech stack . I’ll mark the issue with the question label and see what the other engineers have to say about it.

**Comment by cyfyifanchen** [comment]
Something worth mentioning: we are at the AdventureX 2024 Hangzhou Hackathon. It’s clear that many developers aren’t familiar with the challenges of using Node.js and nvm, especially when dealing with AI, where Python and its environment are predominant.

This makes me think that having a docker-compose file is the right approach. It solves a lot of pain when getting the entire framework up and running.

**Comment by sunshinexcode** [comment]
@Lensual 
docker-compose build astra_playground 会报错, 不能构建出镜像

1.playground/.env.example 后续这个会移除
2..env.sample 这个建议不这样用, docker-compose.yml 改成 docker-compose.yml.example 类似, 使用的时候拷贝一个命名为 docker-compose.yml, 并修改环境变量值

File: playground/Dockerfile
Comment (sunshinexcode) [inline playground/Dockerfile:1]: you can use the following code:

```code
FROM node:20-alpine AS base

FROM base AS builder

WORKDIR /app

COPY . .

RUN npm i && \
    npm run build


FROM base AS runner

WORKDIR /app

ENV NODE_ENV production

RUN mkdir .next

COPY --from=builder /app/.next/standalone ./
COPY --from=builder /app/.next/static ./.next/static

EXPOSE 3000

CMD HOSTNAME="0.0.0.0" node server.js
```

**Comment by cyfyifanchen** [comment]
@Lensual We have concluded that having `docker-compose` is the right thing to do for improving DX. @sunshinexcode will raise a PR for the change. Feel free to review it.

**Comment by sunshinexcode** [comment]
PR: https://github.com/rte-design/ASTRA.ai/pull/123

**Comment by cyfyifanchen** [comment]
@Lensual take a look, if #123 is something you intended to do in the first place? Also from your perspective, which is someone who doesn't know the web development very well, do you think having this `docker-compose` is gonna boost the EX overall?

**Comment by Lensual** [comment]
> @Lensual take a look, if #123 is something you intended to do in the first place? Also from your perspective, which is someone who doesn't know the web development very well, do you think having this `docker-compose` is gonna boost the EX overall?

Yes, it looks good.

**Pull request closed without merging**